app/src/org/commcare/tasks/DataPullTask.java

```diff
@@ -211,6 +213,12 @@ public void reportProgress(int progress) {
                 //Either way, don't re-do this step
                 this.publishProgress(PROGRESS_CLEANED);
 
+                if (isCancelled()) {
+                    // avoid making the http request if user cancelled the task
+                    // NOTE: The result returned is never processed since
+                    // cancelled task results are sent to onCancelled.
+                    return PullTaskResult.UNKNOWN_FAILURE;

```

Comment:
Needs to be updated to 
`return new ResultAndError<>(PullTaskResult.UNKNOWN_FAILURE, "");`